src/components/CreateVmWizard/CreateVmWizard.js

```diff
@@ -149,9 +149,13 @@ class CreateVmWizard extends React.Component {
             clusterId={this.state.steps.basic.clusterId}
             nics={this.state.steps.network.nics}
 
-            onUpdate={({ valid = true, ...updatePayload }) => {
-              this.handleListOnUpdate('network', 'nics', updatePayload)
+            onUpdate={({ valid = true, ...updatePayload }, updateList = true) => {
```

Comment:
@bamsalem - I know you dropped `updateList`, but you didn't make any accommodations for it being empty to avoid rerenders with no deep state updates.

Please add the extra checks I detailed above to `handleListOnUpdate()`.